cmd/test-generation/generate-test.go

```diff
@@ -0,0 +1,342 @@
+// Generates a full netsim test, given a log-splicer processed ssb-fixtures folder and a replication expectations file
+// expectations.json.
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"path"
+	"sort"
+)
+
+func check(err error) {
+	if err != nil {
+		log.Fatalln(err)
+	}
+}
+
+func readExpectations(expectationsPath string) (map[string][]string, error) {
+	b, err := os.ReadFile(expectationsPath)
+	if err != nil {
+		return nil, err
+	}
+	var v map[string][]string
+	err = json.Unmarshal(b, &v)
+	if err != nil {
+		return nil, err
+	}
+	return v, nil
+}
+
+func pickName(splicedFixturesMap map[string]interface{}) string {
+	return splicedFixturesMap["folder"].(string)
+}
+
+// Returns a map of follows (id -> slice of ids that are followed), a map of blocks (isBlocking[id][otherId] = true if id blocks otherId)
+func getFollowMap(followGraphPath string) (map[string][]string, map[string]map[string]bool, error) {
```

Comment:
i feel like the next three functions here deserve a test each, just to make sure to not regress from the format.

Reply:
Good idea :) I'll be breaking up the workhorse code of the programs into modules in the next PR, so that i can combine them into one cohesive workflow. I'll hold off on writing the tests until that is done, less friction :~